### ISSUE TITLE
Test resetting model parameters via `workload.init_model_fn()` during training

### ISSUE THREAD

**Issue by runame** [mlcommons/algorithmic-efficiency#48]
According to the rules, it is expected that the model parameters can be reset by calling `workload.init_model_fn()`. However, creating a new model instance, like it is currently done in the [MNIST PyTorch workload](https://github.com/mlcommons/algorithmic-efficiency/blob/main/workloads/mnist/mnist_pytorch/workload.py#L113-L119), might lead to out-of-memory errors, if the loss/optimizer and potentially other objects are not deleted properly (also described [here](https://discuss.pytorch.org/t/how-to-re-set-alll-parameters-in-a-network/20819/13)). Hence, resetting model paramerters during training should be tested and `workload.init_model_fn()` should potentially be adjusted.
